oauth2_provider/backends.py

```diff
@@ -14,9 +15,14 @@ class OAuth2Backend:
 
     def authenticate(self, request=None, **credentials):
         if request is not None:
-            valid, r = OAuthLibCore.verify_request(request, scopes=[])
-            if valid:
-                return r.user
+            try:
+                valid, request = OAuthLibCore.verify_request(request, scopes=[])
+            except ValueError as error:
+                raise SuspiciousOperation(error)
```

Comment:
As raised in https://github.com/jazzband/django-oauth-toolkit/issues/954#issuecomment-819512062 is catching all `ValueError`s too broad? I think it should be fine, but it might end up exposing error messages users were never meant to see, possibly resulting in security flaws?

Reply:
I'm also a bit worried about this catching "real issues" and causing them to end up not being logged (for example in one Django setup I know `SuspiciousOperation` won't get logged into Sentry but a raw `ValueError` would). So I could imagine some time lost as this absorbs a real error.

The minimalist approach here would be to check if `error.message == "invalid hex string"` (or whatever) and only transform the error then (otherwise just reraising the `ValueError`). But to be honest this might not be a major thing.

Reply:
I've updated it now to check the message of the error, otherwise re-raise the exception as-is.